amethyst_renderer/src/pass/pbm/interleaved.rs

```diff
@@ -103,51 +102,72 @@ where
             mesh_storage,
             tex_storage,
             material_defaults,
-            back_to_front,
+            visibility,
             mesh,
             material,
             global,
             light,
-            transparent,
         ): <Self as PassData<'a>>::Data,
     ) {
         let camera = get_camera(active, &camera, &global);
 
         set_light_args(effect, encoder, &light, &ambient, camera);
 
-        for (mesh, material, global, _) in (&mesh, &material, &global, !&transparent).join() {
-            draw_mesh(
-                encoder,
-                effect,
-                false,
-                mesh_storage.get(mesh),
-                None,
-                &*tex_storage,
-                Some(material),
-                &*material_defaults,
-                camera,
-                Some(global),
-                &[V::QUERIED_ATTRIBUTES],
-                &TEXTURES,
-            );
-        }
-
-        for entity in &back_to_front.entities {
-            if let Some(mesh) = mesh.get(*entity) {
+        match visibility {
+            None => for (mesh, material, global) in (&mesh, &material, &global).join() {
```

Comment:
that line is formatted wrong in all passes basically

Reply:
Nope, that's what rustfmt wants it to be, since it's a single block.